src/libraries/System.Net.Security/src/System/Net/Security/SslStream.cs

```diff
@@ -337,7 +337,7 @@ public virtual void AuthenticateAsServer(X509Certificate serverCertificate, bool
             AuthenticateAsServer(options);
         }
 
-        private void AuthenticateAsServer(SslServerAuthenticationOptions sslServerAuthenticationOptions)
+        public void AuthenticateAsServer(SslServerAuthenticationOptions sslServerAuthenticationOptions)
```

Comment:
I noticed we don't have null checks on these in the async versions.  We should probably add them for both, to throw ArgumentNullException if null is passed in, rather than null ref'ing.

Reply:
Do we have to if the C# null checking feature is enabled? I'll add it, it's not a problem, I'm just curious.

Reply:
Yup.  A few reasons:
1. Consuming apps may not have enabled the analysis.
2. Even if the analysis is enabled, it's not perfect; there are both false positives and false negatives.
3. Even if it does warn and the warning is correct, it can be ignored or suppressed.

Reply:
Added in sync as well as async.